src/redux/actions/loadUser.ts

```diff
@@ -2,11 +2,32 @@ import {Dispatch} from "react";
 import AsyncStorage from "@react-native-async-storage/async-storage";
 import axios, {AxiosError} from "axios";
 import {URI} from "../../URI";
+import {userLoadFailed, userLoadRequest, userLoadSuccess} from "../types/types";
 
-export const loadUser = () => async (dispatch: Dispatch<any>) => {
+interface LoadUserRequestAction {
+    type: 'userLoadRequest';
+}
+
+interface LoadUserSuccessAction {
+    type: 'userLoadSuccess';
+    payload: { user: string, token: string | null };
+}
+
+interface LoadUserFailedAction {
+    type: 'userLoadFailed';
+    payload: string;
+}
+
+type LoadUserAction =
+    | LoadUserRequestAction
+    | LoadUserSuccessAction
+    | LoadUserFailedAction;
+
+
+export const loadUser = () => async (dispatch: Dispatch<LoadUserAction>) => {
```

Comment:
We shouldn't do this "middleware-ish" thing. That ruins flux conceptions. Actions should be fired through dispatch, any work that starts with our redux should be launched in `dispatch`, not by some function that will then call our `dispatch`. Use https://redux.js.org/usage/writing-logic-thunks